cni/network/network.go

```diff
@@ -1451,3 +1468,21 @@ func convertCniResultToInterfaceInfo(result *cniTypesCurr.Result) network.Interf
 
 	return interfaceInfo
 }
+
+func (plugin *NetPlugin) validateArgs(args *cniSkel.CmdArgs, nwCfg *cni.NetworkConfig) error {
+	if !isValidString(args.ContainerID) || !isValidString(args.IfName) {
+		return errors.New("invalid args value")
+	}
+	if !isValidString(nwCfg.Bridge) {
+		return errors.New("invalid network config value")
+	}
+
+	return nil
+}
+
+// returns true if the string fully consists of zero or more alphanumeric, dots, dashes, parentheses, or underscores
+func isValidString(value string) bool {
+	pattern := `^[a-zA-Z0-9._\-\(\) ]*$`
+	re := regexp.MustCompile(pattern)
```

Comment:
It's possible the Go compiler is smart enough to optimize this away, but I doubt it - you should move this to a package var so that it is not being recompiled on every request. The pattern could also be inlined instead of creating an extra intermediate var.